apps/extension/src/background/keyring/service.ts

```diff
@@ -114,6 +114,17 @@ export class KeyRingService {
   ): Promise<AccountStore | false> {
     const publicKeyBytes = fromHex(publicKey);
     const bech32PublicKey = public_key_to_bech32(publicKeyBytes);
+
+    const account = (await this._keyRing.queryAllAccounts()).find(
```

Comment:
Maybe we should separate this piece of code into a new function, once we're having the same behavior in keyring.ts